autospec/tarball.py

```diff
@@ -407,14 +407,15 @@ def set_multi_version(ver):
         # Some build patterns put multiple versions in the same package.
         # For those patterns add to the multi_version list
         if buildpattern.default_pattern in ["godep"]:
-            multi_version.add(ver)
+            multi_version[ver] = ""
         else:
-            multi_version = set([ver])
+            multi_version = {ver:""}
     elif not multi_version:
         # Fall back to ensure a version is always set
         # (otherwise the last known version will be used)
-        multi_version = set("1")
-    return sorted(multi_version)[-1]
+        multi_version[1] = ""
+    latest = sorted(multi_version.keys())[-1]
+    return multi_version[latest]
```

Comment:
Maybe this changes in a later commit but I don't see why latest is being set here.

Does this change the return type to a url rather than a version? That seems off.

Reply:
And looking at the next patch this is resolved, oof this patch series is rough.

Reply:
Yeah, that was an oops. Later commit just returns `latest`.